worldeditadditions_commands/commands/selectors/srect.lua

```diff
@@ -9,39 +9,31 @@ worldedit.register_command("srect", {
 	privs = { worldedit = true },
 	require_pos = 1,
 	parse = function(params_text)
-		local wea = worldeditadditions
+		local wea, vec, tmp = worldeditadditions, vector.new(0, 0, 0), {}
 		local find = wea.split(params_text, "%s", false)
-		local ax1, ax2, len = find[1], find[2], find[table.maxn(find)]
+		local ax1, ax2 = (tostring(find[1]):match('[xyz]') or "g"):sub(1,1), (tostring(find[2]):match('[xyz]') or "g"):sub(1,1)
```

Comment:
Are you sure you need to `tostring()` the items in the `find` array here? `wea.split()` should return a list of strings already.

Reply:
Yes. Those tostrings are 1000% needed to keep everything from getting blown up by a concatenation of nil error if the user provides less that the full number of arguments required by the function. For example when `//srect 2`  parsed by the split function find[1] in an integer and find[2] is nil. When `:match` is called on either of those the resulting type error crashes everything.

Reply:
Yes. Those tostrings are 1000% needed to keep everything from getting blown up by a concatenation of nil error if the user provides less than the full number of arguments required by the function. For example when `//srect 2` parsed by the split function `find[2]` is nil. When `:match` is called on that without it being converted to a string the resulting type error crashes everything.

Reply:
Ahhh I see - got it. Let's keep this as-is then.